### PR TITLE
Fix JupyROOT compilation on Windows

### PR DESCRIPTION
**Pull request by bellenot**

### CODE REVIEW

File: bindings/jupyroot/CMakeLists.txt [hunk 1]
```diff
@@ -42,9 +42,10 @@ foreach(val RANGE ${how_many_pythons})
   # Set the suffix to '.so' and the prefix to 'lib'
   set_target_properties(${libname} PROPERTIES  ${ROOT_LIBRARY_PROPERTIES})
   if(MSVC)
+    list(GET python_libraries ${val} python_library)
     set_target_properties(${libname} PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
     set_target_properties(${libname} PROPERTIES SUFFIX ".pyd")
-    target_link_libraries(${libname} PUBLIC Core)
+    target_link_libraries(${libname} PUBLIC Core  ${python_library})
```
Comment (etejedor) [inline bindings/jupyroot/CMakeLists.txt:48]: So we are linking `libJupyROOT` with `libpython` again. Do you understand why this is necessary for `libJupyROOT` on Windows but not for `libROOTPythonizations` or `libcppyy`? In the end, the three libraries are extension modules that are used from Python: `import libname`.

Comment (bellenot) [reply bindings/jupyroot/CMakeLists.txt:48]: No idea, for now I just want to make it build successfully. I'll check later what is really needed and how to make it fully working on Windows...

**Comment by bellenot** [comment]
> Thanks! Let's also port it to 6.22. We can investigate later a way of building with undefined symbols also on Windows. There must be something missing that we do for `libROOTPythonizations` and `libcppyy`.

You're welcome! Can you cherry pick once I merged it? And one **cannot** build with undefined symbols on Windows. All symbols **must** be defined at link time

**Comment by etejedor** [comment]
> You're welcome! Can you cherry pick once I merged it? And one cannot build with undefined symbols on Windows. All symbols must be defined at link time

Ok I see now how it's done for Windows in the other two libraries. `libROOTPythonizations` is not linked against `libpython`:
https://github.com/root-project/root/blob/master/bindings/pyroot/pythonizations/CMakeLists.txt#L106

but it is linked against `libcppyy`, which links against `libpython` (this part I missed):
https://github.com/root-project/root/blob/master/bindings/pyroot/cppyy/CPyCppyy/CMakeLists.txt#L58

So no undefined symbols for any library indeed :smile: 

Sure I'll do the cherry pick.

**Pull request merged**